src/renderer/pkijs/generatePems/requestCertificate.ts

```diff
@@ -0,0 +1,73 @@
+import { PrintableString, OctetString } from 'asn1js'
+import {
+  CertificationRequest, AttributeTypeAndValue, Extension, Extensions,
+  getCrypto, Attribute
+} from 'pkijs'
+
+import { signAlg, hashAlg } from './config'
+import { generateKeyPair, dumpPEM } from './common'
+
+export const createUserCsr = async (zbayNickanem, commonName, peerId) => {
+  const pkcs10 = await requestCertificate({
+    zbayNickanem: zbayNickanem,
+    commonName: commonName,
+    peerId: peerId,
+    signAlg,
+    hashAlg
+  })
+  await dumpCertificate(pkcs10)
+}
+
+async function requestCertificate({ zbayNickanem, commonName, peerId, signAlg, hashAlg }) {
+  const keyPair = await generateKeyPair({ signAlg, hashAlg })
+  const pkcs10 = new CertificationRequest({
+    version: 0,
+    attributes: []
+  })
+  pkcs10.subject.typesAndValues.push(
+    new AttributeTypeAndValue({
+      type: '2.5.4.3',
```

Comment:
you are adding all 3 attributes (commonName, zbayNickname and peerId) using same OID identifier. The identifier you are using is for CN. This will make it impossible to parse later

You should use different idenfiers for nickname and peerId. We had discussion on #zbay-pl about which OIDs to use.